crates/autopilot/src/run.rs

```diff
@@ -156,6 +171,9 @@ pub async fn run(args: Arguments) {
         );
     }
 
+    // Use unbuffered transport for the Debug API since not all providers support
+    // batched debug calls.
```

Comment:
IMO this is not a good place to explain the issue. Fundamentally the differentiation which web3 instance to use for what matters only inside the `Ethereum` type.